### PR TITLE
Add new settings for clipboard listener

### PR DESCRIPTION
**Pull request by b00f**
Add new settings for Clipboard Listener. It can set to `Auto-detect` or `Timer`.
It addresses #206, However it doesn't fully fix it.

### CODE REVIEW

File: extension.js [hunk 1]
```diff
@@ -557,7 +559,8 @@ const ClipboardIndicator = Lang.Class({
             ExtensionUtils.openPrefs();
         } else {
             Util.spawn([
-                "gnome-shell-extension-prefs",
+                "gnome-extensions",
+                "prefs",
```
Comment (RaphaelRochet) [inline extension.js:563]: This has nothing to do with clipboard listener

Comment (b00f) [reply extension.js:563]: Without this you are not able to ope the setting dialog. For testing the change it is necessary.

Comment (RaphaelRochet) [reply extension.js:563]: Yep. But should not be in PR commit. You could had you 'lister' branch (re)based after that change. You other PR have this lines changed also : accepting one will cause conflict on the other.

Comment (b00f) [reply extension.js:563]: I think this change is related to this PR.

File: prefs.js [hunk 2]
```diff
@@ -91,9 +92,16 @@ const App = new Lang.Class({
         this.field_display_mode = new Gtk.ComboBox({
             model: this._create_display_mode_options()});
 
-        let rendererText = new Gtk.CellRendererText();
-        this.field_display_mode.pack_start (rendererText, false);
-        this.field_display_mode.add_attribute (rendererText, "text", 0);
+        let rendererDsiplayModeText = new Gtk.CellRendererText();
+        this.field_display_mode.pack_start (rendererDsiplayModeText, false);
+        this.field_display_mode.add_attribute (rendererDsiplayModeText, "text", 0);
```
Comment (RaphaelRochet) [inline prefs.js:97]: Dsiplay ? Your fingers seems to have their own spelling :wink: : 
```suggestion
        let rendererDisplayModeText = new Gtk.CellRendererText();
        this.field_display_mode.pack_start (rendererDisplayModeText, false);
        this.field_display_mode.add_attribute (rendererDisplayModeText, "text", 0);
```

**Comment by RaphaelRochet** [comment]
Okay .. so this in fact is a way to force timer mode even if "metaDisplay.get_selection" is not available.
The combo box may induce error on being able to use 'auto-detect'.
Why not use a "Force timer mode (may fix lag issues)" switch ?

**Comment by b00f** [comment]
>Why not use a "Force timer mode (may fix lag issues)" switch ?

I purposely chose combo box. Currently both methods are not perfect. We might find better API or library to track the Clipboard in future.

File: extension.js (hunk 1 above)
Comment (Tudmotu) [reply extension.js:563]: Why did you make this change?

**Comment by Tudmotu** [comment]
@b00f Like we discussed, I think this should be a toggle that is labelled "Use legacy X11 clipboard".
It is:
1. More accurate
2. More obvious that it's related to issues with X11 apps.

**Comment by b00f** [comment]
"legacy X11 clipboard" is a separate API, right?

**Comment by Tudmotu** [comment]
Ok, after looking over the actual code and your changes, you convinced me that the dropdown options are OK. I would say I'm wrong in saying that it's "use x11 clipboard". It's more like "Avoid new GNOME API". But that makes it more confusing to users, so "Auto-Detect" / Timer is better.
But they still feel not ideal. How do they convey the difference, and the problem that it solves?
A user reading the label should be like "ah, that's what I was looking for", and I think these labels don't do that.
Do we have any more ideas for the labels?

**Comment by b00f** [comment]
> Do we have any more ideas for the labels?

How about  having 3 options:
- Auto detect
- Gnome API 
- Using timer

**Comment by b00f** [comment]
Another question is which one should be selected by default. I think timer can set as default value. 

These issues might be related to this PR: #206 #200 #195 
We can make timer as a default option, then monitor the issues. If users doesn't report related issues, we might consider closing theses issues.

**Comment by b00f** [comment]
It's really bad that my PR is still here. If I knew you don't appreciate Open Source culture, I would't work on this application.

**Comment by b00f** [comment]
May I know why you close this PR?

**Comment by Tudmotu** [comment]
@b00f is this still happening to you?

**Comment by b00f** [comment]
I have change my Linux to PopOS!
That was on time I had Arch Linux.
I don't think they have fixed it. The original bug is still open: https://gitlab.gnome.org/GNOME/mutter/-/issues/941

**Pull request closed without merging**